### PR TITLE
feat: Add account level setting for read receipts (WEBAPP-5751)

### PR DESCRIPTION
**Pull request by bennyn**

### CODE REVIEW

File: app/script/properties/PropertiesRepository.js [hunk 1]
```diff
@@ -98,22 +104,43 @@ class PropertiesRepository {
...
+      .getPropertiesByKey(property.key)
+      .then(value => {
+        this.setProperty(property.key, value);
       })
-      .then(() => this._publishProperties());
+      .catch(() => {
+        this.logger.warn(
+          `Property "${property.key}" doesn't exist for this account. Continuing with the default value of "${
+            property.defaultValue
+          }".`
```
Comment (ffflorian) [inline app/script/properties/PropertiesRepository.js]: Can we create a variable with the message beforehand? Makes it more readable :slightly_smiling_face:

File: app/script/user/UserRepository.js [hunk 2]
```diff
@@ -32,21 +32,14 @@ z.user.UserRepository = class UserRepository {
...
-   * @class z.user.UserRepository
-   * @param {z.user.UserService} user_service - Backend REST API user service implementation
-   * @param {z.assets.AssetService} asset_service - Backend REST API asset service implementation
-   * @param {z.self.SelfService} selfService - Backend REST API self service implementation
-   * @param {z.client.ClientRepository} client_repository - Repository for all client interactions
-   * @param {z.time.ServerTimeRepository} serverTimeRepository - Handles time shift between server and client
-   */
-  constructor(user_service, asset_service, selfService, client_repository, serverTimeRepository) {
-    this.user_service = user_service;
+  constructor(user_service, asset_service, selfService, client_repository, serverTimeRepository, propertyRepository) {
```
Comment (ffflorian) [inline app/script/user/UserRepository.js:45]: Why did you remove the JSDoc?

File: app/script/properties/PropertiesRepository.js [hunk 3]
```diff
@@ -98,22 +104,43 @@ class PropertiesRepository {
     return this.selfUser().isTemporaryGuest() ? this._initTemporaryGuestAccount() : this._initActivatedAccount();
   }
 
-  _initActivatedAccount() {
+  _fetchWebAppAccountSettings() {
+    return this.propertiesService
+      .getPropertiesByKey(PropertiesRepository.CONFIG.WEBAPP_ACCOUNT_SETTINGS)
+      .then(properties => {
+        $.extend(true, this.properties, properties);
```
Comment (atomrc) [inline app/script/properties/PropertiesRepository.js:113]: Is the `deep` boolean really important here? 
Else we could remove the call to jquery and use native `Object.assign`

File: app/script/user/UserRepository.js (hunk 2 above)
Comment (bennycode) [reply app/script/user/UserRepository.js:45]: It's back

File: app/script/properties/PropertiesRepository.js (hunk 3 above)
Comment (bennycode) [reply app/script/properties/PropertiesRepository.js:113]: I tried with `Object.assign` but we really need the deep merge here. In the future the whole `PropertiesRepository` will get an overhaul and then we can look again at getting rid off `$.extend`.

File: app/script/properties/PropertiesRepository.js [hunk 4]
```diff
@@ -34,6 +39,7 @@ class PropertiesRepository {
 
     this.properties = new z.properties.PropertiesEntity();
     this.selfUser = ko.observable();
+    this.enableReadReceipts = ko.observable(PropertiesRepository.CONFIG.ENABLE_READ_RECEIPTS.defaultValue);
```
Comment (atomrc) [inline app/script/properties/PropertiesRepository.js:44]: Let's use the `propertiesEntity` that is responsible for holding all the user's properties

File: app/script/properties/PropertiesRepository.js (hunk 3 above)
Comment (atomrc) [reply app/script/properties/PropertiesRepository.js:113]: :+1:

File: app/script/properties/PropertiesRepository.js [hunk 5]
```diff
@@ -168,9 +194,38 @@ class PropertiesRepository {
...
+        break;
+    }
+  }
+
+  // Map a property and set it into our state
+  setProperty(key, value) {
+    switch (key) {
+      case PropertiesRepository.CONFIG.ENABLE_READ_RECEIPTS.key:
+        value = JSON.parse(value);
+        this.logger.info(`Setting key "${key}" to value "${value}".`);
```
Comment (AndyLnd) [inline app/script/properties/PropertiesRepository.js]: Do you want to move this line out of the switch/case? So it will show up for all (future) cases.
